command/agent/http.go

```diff
@@ -336,13 +336,21 @@ func (s *HTTPServer) registerHandlers(enableDebug bool) {
 	s.mux.HandleFunc("/v1/namespace", s.wrap(s.NamespaceCreateRequest))
 	s.mux.HandleFunc("/v1/namespace/", s.wrap(s.NamespaceSpecificRequest))
 
-	if uiEnabled {
+	uiConfigEnabled := s.agent.config.UI != nil && s.agent.config.UI.Enabled
+
+	if uiEnabled && uiConfigEnabled {
 		s.mux.Handle("/ui/", http.StripPrefix("/ui/", s.handleUI(http.FileServer(&UIAssetWrapper{FileSystem: assetFS()}))))
+		s.logger.Debug("UI is enabled")
 	} else {
 		// Write the stubHTML
 		s.mux.HandleFunc("/ui/", func(w http.ResponseWriter, r *http.Request) {
 			w.Write([]byte(stubHTML))
```

Comment:
`stubHTML` is only set when building without the `ui` build tag. This results in the `/ui` endpoint to return a `200` and a empty page.

Looking at Vault and Consul, they each do different things. Vault returns `404` response code and a an error message:
<img width="869" alt="image" src="https://user-images.githubusercontent.com/775380/143131378-21e37182-bc29-47ca-93bf-950b05d38444.png">

Consul returns a `200` and generic content body:
<img width="870" alt="image" src="https://user-images.githubusercontent.com/775380/143131429-1b5f23ae-6e48-4d47-b3ca-213362784336.png">

Personally I would say Vault's seems to be the most correct approach, but no strong feelings. I think we could move this handler registration to the `else` clause below and add a generic `404` body content in the [`handleRootFallthrough`](https://github.com/hashicorp/nomad/blob/v1.2.1/command/agent/http.go#L420) handler.

Reply:
Yeah I'd intentionally not touched the existing behavior for the fallthrough, but a 404 does sound a lot nicer. Will do.

Reply:
Actually, having tried it just now I don't think that's very user friendly. If you go to `http://localhost:4646` you end up getting a redirect to `http://localhost:4646/ui` with just the browser's 404 page. Having the stub content there and filling it out so that it does something more similar to Consul seems nicer.

Reply:
Fixed in 8006bd6